crates/example-types/src/node_types.rs

```diff
@@ -137,8 +137,8 @@ impl<TYPES: NodeType> NodeImplementation<TYPES> for Libp2pImpl {
 pub struct TestVersions {}
 
 impl Versions for TestVersions {
-    type Base = StaticVersion<0, 1>;
-    type Upgrade = StaticVersion<0, 2>;
+    type Base = StaticVersion<0, 3>;
```

Comment:
It would be nice to test the upgrade from v2 to v3 here :)

Reply:
This was just for testing -- I wanted to make it easy to run the examples off the branch. I'll be reverting this before merging, and relying on integration tests (rather than the examples)